pkg/transforms/policy.go

```diff
@@ -146,51 +146,72 @@ func recordRelatedObjects(c *unstructured.Unstructured, node Node) Node {
 	}
 
 	objList := make([]relatedObject, 0, len(relatedObjects))
+	nonCompliantList := []string{}
 
 	for _, item := range relatedObjects {
-		relObj := parseConfigPolicyRelatedObject(item)
+		relObj, compliant := parseConfigPolicyRelatedObject(item)
 		if relObj == nil {
 			continue
 		}
 
 		objList = append(objList, *relObj)
+
+		if !compliant {
+			nonCompliantList = append(nonCompliantList, relObj.String())
+		}
 	}
 
 	slices.SortFunc(objList, func(a, b relatedObject) int {
-		return strings.Compare(a.kind+a.namespace+a.name, b.kind+b.namespace+b.name)
+		return strings.Compare(a.String(), b.String())
 	})
 
+	slices.Sort(nonCompliantList)
+
 	node.Metadata["relObjs"] = objList
+	node.Properties["nonCompliantObjects"] = strings.Join(nonCompliantList, "; ")
```

Comment:
Why storing in node.Properties?
Ideally we should store this data as an edge with `EdgeType = nonCompliant`.  **However, we currently can't query by edgeType, so we'd need to work on that limitation.**

Reply:
How many nonCompliant objects do you expect to have on a typical policy? Is there a max?
I'm worried about this string getting too big.

Reply:
There's no hard maximum, and it's hard to say what a "typical" policy might have. Because we allow templating, loops, and complicated selectors in our ConfigurationPolicy syntax, it could potentially be a large number. Of course, the idea would be that a non-compliant resource would need to be remediated, so the "normal" state for resources might be compliant.

Reply:
`EdgeType = nonCompliant` would be nice, but it wouldn't help inform users about resources that are desired, but completely missing.

Reply:
The `argoapplication.go` has a very similar concept. Please check that transform and let's try to keep it as similar as possible.
In the Argo transform, if the resource exists, we build and edge and for missing resources we store it in `_missingResources`.  [Check the logic here.](https://github.com/stolostron/search-collector/blob/main/pkg/transforms/argoapplication.go#L200-L219)